atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/workflow/MultipleBusyCellWorkflowsTest.java

```diff
@@ -0,0 +1,188 @@
+                        .map(x -> x.accept(new WitnessedTransactionActionVisitor<WorkloadCell>() {
+
+                            @Override
+                            public WorkloadCell visit(WitnessedSingleCellReadTransactionAction readTransactionAction) {
+                                return readTransactionAction.cell();
+                            }
+
+                            @Override
+                            public WorkloadCell visit(WitnessedWriteTransactionAction writeTransactionAction) {
+                                return writeTransactionAction.cell();
+                            }
+
+                            @Override
+                            public WorkloadCell visit(WitnessedDeleteTransactionAction deleteTransactionAction) {
+                                return deleteTransactionAction.cell();
+                            }
+
+                            @Override
+                            public WorkloadCell visit(
+                                    WitnessedRowColumnRangeReadTransactionAction rowColumnRangeReadTransactionAction) {
+                                throw new RuntimeException("We don't expect to see this action");
+                            }
+                        }))
+                        .collect(Collectors.toSet()))
+                .hasSizeBetween(1, MAX_CELLS);
+    }
+
+    @Test
+    public void transactionWriteActionsConvergeToConfiguredProbabilities() {
+        WorkflowHistory workflowHistory = workflow.run();
+        AtomicInteger writes = new AtomicInteger();
+        AtomicInteger deletes = new AtomicInteger();
+        workflowHistory.history().stream()
+                .flatMap(x -> x.actions().stream())
+                .forEach(x -> x.accept(new WitnessedTransactionActionVisitor<Void>() {
+
+                    @Override
+                    public Void visit(WitnessedSingleCellReadTransactionAction readTransactionAction) {
+                        return null;
+                    }
+
+                    @Override
+                    public Void visit(WitnessedWriteTransactionAction writeTransactionAction) {
+                        writes.incrementAndGet();
+                        return null;
+                    }
+
+                    @Override
+                    public Void visit(WitnessedDeleteTransactionAction deleteTransactionAction) {
+                        deletes.incrementAndGet();
+                        return null;
+                    }
+
+                    @Override
+                    public Void visit(
+                            WitnessedRowColumnRangeReadTransactionAction rowColumnRangeReadTransactionAction) {
+                        throw new RuntimeException("We don't expect to see this action");
+                    }
+                }));
+        double posteriorProbability = deletes.get() / (double) (writes.get() + deletes.get());
+        assertThat(posteriorProbability).isCloseTo(DELETE_PROBABILITY, within(0.05));
```

Comment:
cell count is 20000 (1000 per cell * 20 cells). 18000 of these are writes/deletes - yeah, it would be pretty extreme if you had fewer than 2700 or more than 4500